### PR TITLE
Strict node

### PR DESCRIPTION
**Pull request by yoozo**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Enable TS strict model in node package

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have tested locally
- [ ] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch
- [x] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

File: packages/node/src/configure/SubqueryProject.ts [hunk 1]
```diff
@@ -73,7 +69,7 @@ export class SubqueryProject implements ISubqueryProject {
     this.#dataSources = await insertBlockFiltersCronSchedules(
       this.dataSources,
       getTimestamp,
-      isRuntimeDs,
+      isRuntimeDs, // TODO need update main repo to support this
```
Comment (stwiname) [inline packages/node/src/configure/SubqueryProject.ts:72]: I don't understand this comment, what needs to change?

File: packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts [hunk 2]
```diff
@@ -146,15 +148,15 @@ function eventFilterToDictionaryCondition(
 function sanitiseDictionaryConditions(
   dictionaryConditions: EthDictionaryV2QueryEntry,
 ): EthDictionaryV2QueryEntry {
-  if (!dictionaryConditions.logs.length) {
+  if (dictionaryConditions.logs?.length === 0) {
```
Comment (stwiname) [inline packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts:151]: I think this is a behaviour change. If `logs` is undefined then it wont be removed.

```suggestion
  if (!dictionaryConditions.logs?.length) {
```

File: packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts [hunk 3]
```diff
@@ -146,15 +148,15 @@ function eventFilterToDictionaryCondition(
...
     delete dictionaryConditions.logs;
   } else {
     dictionaryConditions.logs = uniqBy(dictionaryConditions.logs, (log) =>
       JSON.stringify(log),
     );
   }
 
-  if (!dictionaryConditions.transactions.length) {
+  if (dictionaryConditions.transactions?.length === 0) {
```
Comment (stwiname) [inline packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts:159]: Same as above comment

File: packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts [hunk 4]
```diff
@@ -192,7 +194,7 @@ export function buildDictionaryV2QueryEntry(
             filter.to !== undefined ||
             filter.function !== undefined
           ) {
-            dictionaryConditions.transactions.push(
+            dictionaryConditions.transactions?.push(
```
Comment (stwiname) [inline packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts:197]: Should `dictionaryConditions.transactions` be defined if not?

Same applies to logs further down.

File: packages/node/src/indexer/dictionary/v2/types.ts [hunk 5]
```diff
@@ -78,7 +78,7 @@ export interface EthDictionaryLogConditions {
 }
 
 export interface EthDictionaryTxConditions {
-  to?: string[];
+  to?: (string | null)[];
```
Comment (stwiname) [inline packages/node/src/indexer/dictionary/v2/types.ts:81]: I don't think this is right. The implementation is probably incorrect if null can be part of the array

File: packages/node/src/indexer/fetch.service.ts [hunk 6]
```diff
@@ -86,7 +86,7 @@ export class FetchService extends BaseFetchService<
   }
 
   protected async initBlockDispatcher(): Promise<void> {
-    await this.blockDispatcher.init(this.resetForNewDs.bind(this));
+    await this.blockDispatcher.init(this.resetForNewDs.bind(this)); // TODO need update main repo to support this
```
Comment (stwiname) [inline packages/node/src/indexer/fetch.service.ts:89]: What needs to be done on the main repo?

File: packages/types/src/ethereum/interfaces.ts [hunk 7]
```diff
@@ -22,7 +22,7 @@ export interface EthereumTransactionFilter {
    * @example
    * to: '0x220866B1A2219f40e72f5c628B65D54268cA3A9D',
    **/
-  to?: string;
+  to?: string | null;
```
Comment (stwiname) [inline packages/types/src/ethereum/interfaces.ts:25]: I don't think this needs to change. Code in the node can be updated instead
```suggestion
  to?: string;
```

File: packages/node/src/configure/SubqueryProject.ts (hunk 1 above)
Comment (yoozo) [reply packages/node/src/configure/SubqueryProject.ts:72]: The subql repo needs to be released first, then we can update the dependencies on our end.

File: packages/node/src/indexer/dictionary/v2/types.ts (hunk 5 above)
Comment (yoozo) [reply packages/node/src/indexer/dictionary/v2/types.ts:81]: There are currently many checks for to being equal to null in the code.

File: packages/types/src/ethereum/interfaces.ts (hunk 7 above)
Comment (stwiname) [reply packages/types/src/ethereum/interfaces.ts:25]: I think this still needs to be only optional string. The dictionary code should change if its wrong

Comment (yoozo) [reply packages/types/src/ethereum/interfaces.ts:25]: image I see that the test case includes this usage scenario. Shouldn’t we support assigning null?
image

**Review by stwiname — APPROVED** [review]

**Pull request merged**